client.go

```
@@ -186,9 +189,13 @@ func (c *Client) CloseIdleConnections() {
 	c.HTTPClient.Transport.(connectionCloser).CloseIdleConnections()
 }
 
-func (c *Client) setTokenHeader(r *http.Request) {
-	c.Token.GenerateIfExpired()
+func (c *Client) setTokenHeader(r *http.Request) error {
+	_, err := c.Token.GenerateIfExpired()
+	if err != nil {
+		return err
```

Behavior changes. The default may be better to ignore errors as before.